app/controllers/api/red_hat_cloud_service_providers_controller.rb

```diff
@@ -0,0 +1,40 @@
+module Api
+  class RedHatCloudServiceProvidersController < BaseController
+    def red_hat_cloud_service_providers_search_conditions
+      { :type => provider_types }
+    end
+
+    def find_red_hat_cloud_service_providers(id)
+      collection_type = :red_hat_cloud_service_providers
+      klass = collection_class(collection_type)
+      provider = find_resource(klass, "id", id)
+      provider = filter_resource(provider, collection_type, klass)
+      raise NotFoundError, "Invalid #{provider_title} id:#{id} specified" unless provider_types.include?(provider.type)
+
+      provider
+    end
+
+    def sync_resource(_type, id, _data)
+      provider = find_red_hat_cloud_service_providers(id)
+      desc = "Syncing #{provider_ident(provider)}"
+      task_id = Cfme::CloudServices::InventorySync.sync_queue(User.current_user.userid, provider)
+      action_result(true, desc, :task_id => task_id)
+    rescue => e
+      action_result(false, e.to_s)
+    end
+
+    private
+
+    def provider_ident(provider)
+      "#{provider_title} id:#{provider.id} name:'#{provider.name}'"
+    end
+
+    def provider_title
+      "Red Hat Cloud Services Provider"
```

Comment:
I don't understand the provider_title.  The providers should just be the provider.type (or more specifically whatever the human description of the provider type is).  There's not such thing as a "Red Hat Cloud Services Provider"

Reply:
plain old "Provider" should do the trick here.  it's just the displayname representing the kind of resources the collection endpoint returns.